src/lib/registration/ui/regservice_selection_dialog.rb

```diff
@@ -0,0 +1,91 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "ui/dialog"
+require "registration/helpers"
+require "slp/dialogs/service_selection"
+
+module Registration
+  module UI
+    # This class implements a SCC/SMT service selection dialog.
+    class RegserviceSelectionDialog < Yast::Dialogs::ServiceSelection
+      Yast.import "UI"
+      Yast.import "Label"
+      Yast.import "Report"
+
+      # Run dialog
+      #
+      # The return value will be:
+      # * A service in case one SMT server was selected
+      # * :scc symbol if default SCC was selected
+      # * :cancel symbol if the SCC was canceled (pressing the 'cancel' button)
+      #
+      # @example Select the default SCC service
+      #   Registration::UI::SelectionServiceDialog.run(services) #=> :scc
+      #
+      # @example Select some SMT service
+      #   Registration::UI::SelectionServiceDialog.run(services)
+      #     #=> #<Yast::SlpServiceClass::Service...>
+      #
+      # @param services [Array<SlpServiceClass::Service] list of services to show
+      # @return [SlpServiceClass::Service,Symbol] selected service or symbol (:scc or :cancel)
+      #
+      # @see #run
+      def self.run(services:, heading: nil, description: nil)
+        new(services: services, heading: heading, description: description).run
+      end
+
+      # Constructor
+      #
+      # @param services [Array<SlpServiceClass::Service] list of services to show
+      def initialize(services: [], heading: nil, description: nil)
+        super(
+          services:        [scc_service] + services,
+          heading:         heading || _("Local Registration Servers"),
+          description:     description || _("Select a detected registration server " \
+                             "from the list\nor the default SUSE registration server."),
+          no_selected_msg: _("No registration server selected")
+        )
+      end
+
+      # Handler for the Ok button
+      #
+      # If no option was selected, a error message is shown.
+      #
+      # @see Yast::Dialogs::ServiceSelection#ok_handler
+      def ok_handler
+        selected = super
+        finish_dialog(:scc) if selected == scc_service
+      end
+
+      # Return the service description to be shown to the user
+      #
```

Comment:
I expect here to documented parameter as it looks like super method do not handle string at all? so what can be passed?